zcash_client_sqlite/src/wallet.rs

```diff
@@ -1919,50 +1934,45 @@ pub(crate) fn get_account<P: Parameters>(
     params: &P,
     account_uuid: AccountUuid,
 ) -> Result<Option<Account>, SqliteClientError> {
-    let mut sql = conn.prepare_cached(
+    let mut stmt = conn.prepare_cached(
         r#"
-        SELECT name, account_kind, hd_seed_fingerprint, hd_account_index, key_source, ufvk, uivk, has_spend_key
+        SELECT name, uuid, account_kind,
+               hd_seed_fingerprint, hd_account_index, key_source,
+               ufvk, uivk, has_spend_key
         FROM accounts
         WHERE uuid = :account_uuid
         "#,
     )?;
 
-    let mut result = sql.query(named_params![":account_uuid": account_uuid.0])?;
-    let row = result.next()?;
-    match row {
-        Some(row) => {
-            let account_name = row.get("name")?;
-            let kind = parse_account_source(
-                row.get("account_kind")?,
-                row.get("hd_seed_fingerprint")?,
-                row.get("hd_account_index")?,
-                row.get("has_spend_key")?,
-                row.get("key_source")?,
-            )?;
+    let mut rows = stmt.query_and_then::<_, SqliteClientError, _, _>(
+        named_params![":account_uuid": account_uuid.0],
+        |row| parse_account_row(row, params),
+    )?;
 
-            let ufvk_str: Option<String> = row.get("ufvk")?;
-            let viewing_key = if let Some(ufvk_str) = ufvk_str {
-                ViewingKey::Full(Box::new(
-                    UnifiedFullViewingKey::decode(params, &ufvk_str[..])
-                        .map_err(SqliteClientError::BadAccountData)?,
-                ))
-            } else {
-                let uivk_str: String = row.get("uivk")?;
-                ViewingKey::Incoming(Box::new(
-                    UnifiedIncomingViewingKey::decode(params, &uivk_str[..])
-                        .map_err(SqliteClientError::BadAccountData)?,
-                ))
-            };
+    rows.next().transpose()
+}
 
-            Ok(Some(Account {
-                name: account_name,
-                uuid: account_uuid,
-                kind,
-                viewing_key,
-            }))
-        }
-        None => Ok(None),
-    }
+pub(crate) fn get_account_internal<P: Parameters>(
```

Comment:
Not clear to me why this is added in the second commit, and then deleted in the third commit.